scripts/fastapi/cache_dumper.py

```
@@ -127,104 +139,178 @@ def batch_write_to_dynamodb(table_name: str, items: List[Dict]) -> Dict:
+                        UpdateExpression=f"SET {', '.join(update_expr_parts)}",
+                        ExpressionAttributeValues=expr_attr_values
+                    )
+
+                total_synced += 1
+
+                # Update checkpoint after successful operation to avoid replay
+                entry_sequence = entry.get('sequence', -1)
+                if entry_sequence >= 0:
+                    wal_manager.set_last_applied_sequence(entry_sequence)
+
+            except Exception as e:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)